src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs

```diff
@@ -213,27 +213,34 @@ public RunspaceDetails CurrentRunspace
                     hostUserInterface,
                     logger);
 
-            Runspace initialRunspace = PowerShellContextService.CreateRunspace(psHost, hostStartupInfo.LanguageMode);
+            if (hostStartupInfo.InitialSessionState.LanguageMode != PSLanguageMode.FullLanguage)
+            {
+                hostStartupInfo.InitialSessionState.ImportPSModule(hostStartupInfo.AdditionalModules.ToArray());
+                hostStartupInfo.InitialSessionState.ImportPSModule(new string[] { s_commandsModulePath });
+            }
+            Runspace initialRunspace = PowerShellContextService.CreateRunspace(psHost, hostStartupInfo.InitialSessionState);
             powerShellContext.Initialize(hostStartupInfo.ProfilePaths, initialRunspace, true, hostUserInterface);
-
-            powerShellContext.ImportCommandsModuleAsync();
-
             // TODO: This can be moved to the point after the $psEditor object
             // gets initialized when that is done earlier than LanguageServer.Initialize
-            foreach (string module in hostStartupInfo.AdditionalModules)
+            // Darren Kattan: I haven't tested it, but I have a feeling this entire bit of logic can be replaced with the bit above for non-FullLanguage mode. I think that was is cleaner anyway.            
```

Comment:
I think my concerns with using ImportPSModule is that I'm not sure the user will get the output if their module doesn't import correctly.

Thats why I think we just used Install-Module before.

I don't have enough data to say that they won't get an error with ImportPSModule so if you happen to know that they would, then I say move this up.

Otherwise, move it all back to Import-Module.

Import-Module will gracefully handle when the path to the module doesn't exist (with an error in the console)

Reply:
After extensive testing, it turns out that the _only_ way to get the required modules to load properly in a Constrained Runspace (specifically PowerShellEditorServices.Commands and other modules that need access to the file system at load time) is to use InitialSessionState.ImportPSModuleFromPath

Loading it with ImportPSModule, or using Import-Module from within the already opened Runspace both fail with System.Management.Automation.DriveNotFoundException: 'Cannot find drive. A drive with the name 'C' does not exist.'